component-classes/index.js

```diff
@@ -93,7 +93,7 @@ export const pill = {
   label: 'pl-12 py-8 rounded-l-full',
   labelWithoutClose: 'pr-12 rounded-r-full',
   labelWithClose: 'pr-2',
-  close: 'pr-12 pl-4 py-10 rounded-r-full',
+  close: 'pr-12 pl-4 pt-4 pb-6 rounded-r-full !text-m',
```

Comment:
I am (again) worried about our releases here...
If we release this, it will be a breaking change. Consumers using older versions of our packages with the svg solution will get wrong paddings.
So I guess we need keep the old `close` here, and add another one (`closeX`? 🤷) to use for the x-character solution?
Or does anyone else see another solution to this? 😅

Reply:
...or am I just confused and this will be baked in to the component packages with the right versions? 😅  I guess so? Or? 🙈 😅

Reply:
...could it then be that all needed classes for the components in different versions is not included in `https://assets.finn.no/pkg/@warp-ds/css/v1/components.css` ? 🤔

Reply:
Hmm good questions. I'm not sure. Maybe best if I just add another one `closeX` to be on the safe side? :)

Reply:
It looks like with these changes 👇 syncing "externalized" `classes` array and `components.css` will no longer be a problem, so maybe we could add a `TODO` to remove redundant `ccPill.close` key when we think the below changes are implemented in most apps? Just so we don't get scared to clean things up in the future. Wdyt?
https://github.com/warp-ds/drive/pull/160
https://github.com/warp-ds/css/pull/56

Reply:
I'll try to sum up what we've discovered through testing various approaches with Felicia regarding the initial concern you raised @magnuh.
**tldr; we think it's fine to update `pill.close` with new classes. Most users will test their apps when updating our packages, hopefully, and Pill seems to only be used by 1 team that doesn't even seem to rely on `safelist`.**

1. We can update the `pill.close` key with new padding classes. No need to have backwards compatibility. 
2. When this change is applied in component packages (vue/react), we will need to update @warp-ds/css to the newest version. The new version of vue/react will then have the new classes in pill close button.
3. When installing that new vue/react version in a project, and that project relies on `safelist` in `uno.config.js` to apply styles to components, @warp-ds/css will also need to be updated to the newest version. This way we ensure the new classes (`pt-4 pb-6`) are included in `uno.css` file. Here we'd be having a breaking change if users only update vue/react, and not @warp-ds/css. BUT we can always let users know in the release notes (merge commit message) that if they are using @warp-ds/css to generate component classes with UnoCSS, that dep would need to be updated to a certain version. As far as I checked, there's only [one team](https://github.schibsted.io/finn/product-podium/blob/f971f9e43247f5426f022e80a8e7a6cb69e1ddd2/podlets/product-price-info-podlet/src/client/components/EffectPriceCalculator.tsx) that uses Pill for now so we could tell them to test things before updating @warp-ds/react.

Finally, we are moving away from `safelist` now that `components.css` can be the source of components' styling. And with the PRs linked above that css file will always be in sync with the components classes array that is used to optimise `uno.css` (so UnoCSS omits component classes in the process).

Reply:
Thinking out loud (about backwards compatibility) now... 😅
 
- The `component.css` is updated on the page level with these new changes (which will **not** include the `py-10` class).
- There is a team still using the old component with close svg (which requires the `py-10` class).

Will uno then pick up the missing py-10 class and create css for it in their build... or will it be missed? 🤔 

(The above is just an example, and I understand that it might not be a problem in this exact case. 😄 And, again, I might be totally off here and it will not be a problem. 🤷 😅)

Reply:
If I remember correctly we saw that py-10 is still being used for other css-classes in the component-classes so I'm assuming that uno will anyways create css for it in the build? @BalbinaK correct me if I've misunderstood this. 😄

Reply:
But I guess that it potentially would be a problem in other situations if the removed classes are not used on anything else, no? 😅

Reply:
Hmm I guess, but in this case it shouldn’t be an issue 🤔 🙈

Reply:
> The component.css is updated on the page level with these new changes (which will not include the py-10 class).
There is a team still using the old component with close svg (which requires the py-10 class).

As Felicia explain. I forgot to mention in my summary that `py-10` is also used [here](https://github.com/warp-ds/css/blob/next/component-classes/index.js#L270), which means components.css will still have this class handled if old versions of @warp-ds/react are relying on [components.css](https://assets.finn.no/pkg/@warp-ds/css/v1/components.css).

Reply:
Yes, as I tried to explain, I totally understand that we are lucky here that `py-10` exists elsewhere. 🍀 😄 
...my concern is that we can't really rely on luck every time we need to do these kind of changes? 😅 ...and I would just want to understand "the correct" and "future proof" way of doing this. 🙂 
But I guess we can wait until the next time to solve this. 👍 🚣‍♂️

Reply:
If we would end up in a similar scenario in the future where a css-class would no longer be in use, I guess we should maybe then do a similar solution to what you proposed previously, that we would have to create a separate css attribute, e.g. ccPill.closeX. But yes we should probably discuss or way of working if a css-class is removed and no longer in use. 😄